test/e2e/functional_test.go

```diff
@@ -0,0 +1,368 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	kapierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	watchtools "k8s.io/client-go/tools/watch"
+	"k8s.io/kubernetes/pkg/client/conditions"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const waitForPrometheusStartSeconds = 240
+
+var _ = g.Describe("[Feature:Prometheus][Conformance] Prometheus", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithoutNamespace("prometheus")
+
+		url, bearerToken string
+	)
+
+	g.BeforeEach(func() {
+		var ok bool
+		url, bearerToken, ok = locatePrometheus(oc)
+		if !ok {
+			e2e.Skipf("Prometheus could not be located on this cluster, skipping prometheus test")
+		}
+	})
+
+	g.Describe("when run on the cluster", func() {
+
+		oc = exutil.NewCLI("monitoring", exutil.KubeConfigPath())
+
+		g.It("prometheus & alertmanager pods should use anti-affinity", func() {
+			oc.SetNamespace("openshift-monitoring")
+			e2e.Logf("Add admin role to current user.")
+			err := oc.AsAdmin().Run("adm").Args("policy", "add-role-to-user", "admin", oc.Username()).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			e2e.Logf("Get list of pods for specific namespace.")
+			podList, err := oc.AdminKubeClient().CoreV1().Pods(oc.Namespace()).List(metav1.ListOptions{})
+
+			if err != nil {
+				e2e.Logf("Error in podList: %v", err)
+				return
+			}
+
+			var alm = `affinity:
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - podAffinityTerm:
+          labelSelector:
+            matchExpressions:
+            - key: alertmanager
+              operator: In
+              values:
+              - main`
+
+			var k8s = `affinity:
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - podAffinityTerm:
+          labelSelector:
+            matchExpressions:
+            - key: prometheus
+              operator: In
+              values:
+              - k8s`
+
+			for _, pod := range podList.Items {
+				if strings.Contains(pod.Name, "alertmanager-main") {
+					e2e.Logf("Try to get %s pod config.", pod.Name)
+					podYamlOutput, err := oc.Run("get").Args("pod", pod.Name, "-o", "yaml").Output()
+					if err != nil {
+						e2e.Logf("Error with getting pod yaml: %s\n", err)
+					}
+					e2e.Logf("Check that pod config include proper ani-affinity section.")
```

Comment:
```suggestion
					e2e.Logf("Check that pod config include proper anti-affinity section.")
```

I think this was anti?